composer/loggers/mosaicml_logger.py

```diff
@@ -96,10 +125,104 @@ def log_hyperparameters(self, hyperparameters: Dict[str, Any]):
+            get_logger_type(logger) if not isinstance(logger, MosaicMLLogger) else 'MosaicMLLogger'
+            for logger in loggers
+        ]
+
+        # Take the service provider out of the URI and log it to metadata. If no service provider
+        # is found (i.e. backend = ''), then we assume 'local' for the cloud provider.
+        if load_path is not None:
+            backend, _, _ = parse_uri(load_path)
+            metrics['composer/cloud_provider_data'] = backend if backend else 'local'
+        if save_folder is not None:
+            backend, _, _ = parse_uri(save_folder)
+            metrics['composer/cloud_provider_checkpoints'] = backend if backend else 'local'
+
+        # Save interval can be passed in w/ multiple types. If the type is a function, then
+        # we log 'callable' as the save_interval value for analytics.
+        if isinstance(save_interval, Union[str, int]):
+            save_interval_str = str(save_interval)
+        elif isinstance(save_interval, Time):
+            save_interval_str = f'{save_interval._value}{save_interval._unit}'
+        else:
+            save_interval_str = 'callable'
+        metrics['composer/save_interval'] = save_interval_str
+
+        if trainer_state.fsdp_config:
+            metrics['composer/sharding_strategy'] = trainer_state.fsdp_config.get('sharding_strategy', None)
+            metrics['composer/activation_checkpointing'] = trainer_state.fsdp_config.get(
+                'activation_checkpointing',
+                False,
+            )
+            metrics['composer/forward_prefetch'] = trainer_state.fsdp_config.get('forward_prefetch', False)
+            metrics['composer/backward_prefetch'] = trainer_state.fsdp_config.get(
+                'backward_prefetch',
+                FSDP(Module()).backward_prefetch,
+            )
+
+            # Get device_mesh from config so it is in list form and JSON parsable
+            metrics['composer/device_mesh'] = trainer_state.fsdp_config.get('device_mesh', [])
+
+            mixed_precision = trainer_state.fsdp_config.get('mixed_precision', None)
+            if mixed_precision is not None and isinstance(mixed_precision, dict):
+                # Sorting the keys allows us to parse this dict value as JSON in a SQL query if needed
+                metrics['composer/mixed_precision'] = json.dumps(mixed_precision, sort_keys=True)
+            else:
+                metrics['composer/mixed_precision'] = mixed_precision
+
+        if trainer_state.fsdp_state_dict_type is not None:
+            metrics['composer/state_dict_type'] = trainer_state.fsdp_state_dict_type
+
+        self.log_metrics(metrics)
+        self._flush_metadata(force_flush=True)
+
     def log_exception(self, exception: Exception):
         self._log_metadata({'exception': exception_to_json_serializable_dict(exception)})
         self._flush_metadata(force_flush=True)
 
+    def init(self, state: State, logger: Logger) -> None:
+        self.log_analytics()
```

Comment:
state should just get passed along here. It does not need to be in the stashed analytics dictionary